lib/v2/screens/explore_screens/vote/interactor/mappers/proposals_state_mapper.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:seeds/v2/datasource/remote/model/proposals_model.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/v2/i18n/explore_screens/invite/invite.i18n.dart';
+import 'package:seeds/v2/screens/explore_screens/vote/interactor/viewmodels/vote_state.dart';
+
+class ProposalsStateMapper extends StateMapper {
+  VoteState mapResultToState(VoteState currentState, Result result) {
+    if (result.isError) {
+      return currentState.copyWith(pageState: PageState.failure, errorMessage: "Error loading proposals".i18n);
+    } else {
+      ProposalsModel proposalsModel = result.asValue!.value as ProposalsModel;
+      Map<String, String> currentType = currentState.currentType!;
```

Comment:
how do you know current type is not null? Avoid using (!) see if you can mke currentType not nullable

Reply:
because you need move the tab in order to trigger a call. (for that moment it already has a value).

Reply:
But now rethinking in my spare time. I'm sure needs the default value for the first secction load  😆